LedMatrix/PowerShell/SenseHat.Matrix/SenseHat.Matrix.psm1

```diff
@@ -0,0 +1,246 @@
+    }
+
+    $return
+}
+
+#####
+# Public functions
+
+Function Set-MatrixWithRainbow {
+    $Color = [UInt16]32768
+    [UInt16[]]$PixelList = $Color
+    1..15 | foreach {
+        $PixelList += $Color -bor ($Color -shr 1)
+        $Color = $PixelList[-1]
+    }
+    1..16 | foreach {
+        $PixelList += $Color -shr 1
+        $Color = $PixelList[-1]
+    }
+    
+    [UInt16[]]$PixelList += $PixelList[($PixelList.Length -1)..0]
+    
+    $PixelListByte = ConvertToByteArray -source $PixelList
+    WritebyteArrayToMatrix -PixelList $PixelListByte
+}
+
+Function Set-MatrixWithSingleColor { 
+    Param(
+        [Parameter(Mandatory = $true, 
+            ValueFromPipeline = $true, 
+            ValueFromPipelineByPropertyName = $true, 
+            Position = 0
+        )]
+        [ValidateRange(0,31)]
+        [UInt16]$Red,
+
+        [Parameter(Mandatory = $true, 
+            ValueFromPipeline = $true, 
+            ValueFromPipelineByPropertyName = $true, 
+            Position = 0
+        )]
+        [ValidateRange(0,63)]
+        [UInt16]$Green,
+
+        [Parameter(Mandatory = $true, 
+            ValueFromPipeline = $true, 
+            ValueFromPipelineByPropertyName = $true, 
+            Position = 0
+        )]
+        [ValidateRange(0,31)]
+        [UInt16]$Blue
+    )
+
+    Begin {
+        $PixelListInt = @()
+    }
+
+    Process {
+        for ($I = 0 ; $I -lt 64 ; $I++ ) {
+            [UInt16[]]$PixelListInt += [UInt16]( ([UInt16]$Red -shl 11) -bor ([UInt16]$Green -shl 5) -bor ([UInt16]$Blue) )
+        }
+
+        Try {
+            $PixelListByte = ConvertToByteArray -source $PixelListInt
+            WritebyteArrayToMatrix -PixelList $PixelListByte
+        }
+        catch {
+            Write-Error 'Failed to write output to screen.'
+        }
+    }
+}
+
+Function Write-SenseHatMatrix {
+    Param(
+        [Parameter(Mandatory=$True)]
+        [ValidateNotNullOrEmpty()]
+        [String]$Text,
+
+        [String]$Font = 'cherry-10-b',
+        [UInt16]$BackgroundColor = 0,
+        [UInt16]$ForegroundColor = 63488,
+        [Int]$TextSpeed = 1,
+        [Int]$Loop = 1
+    )
+
+    $FontCache = GetAvailableFonts | Where-Object -Property Name -eq $Font | ParseBitmapFont
+    $Pages = ConvertTextToByteArray -Text $Text -BitmapFont $FontCache -ForeGroundColor $ForegroundColor -BackGroundColor $BackgroundColor
+    
+    1..$Loop | Foreach { 
+        foreach ($Page in $Pages) {
+            $ByteArray = ConvertToByteArray -source $Page
+            WritebyteArrayToMatrix -PixelList $ByteArray
+            Start-Sleep -Seconds $TextSpeed
+        }
+    }
+    Set-MatrixWithSingleColor -Red 0 -Green 0 -Blue 0    
+}
+
+Function Get-RGB565Color {
+    param(
+        [Parameter(Mandatory=$True)]
+        [ValidateSet('Red','Green','Blue','Yellow','Orange','Purple','Pink','LightBlue','LightRed','LightGreen','LightYellow','DarkBlue','DarkRed','DarkGreen','DarkYellow','grey','white','black','darkpink','DarkOrange')]
```

Comment:
I _think_ that this can work with the Color enum (because it also has RGB) . One thing that I thought some time ago but had no luck, was that we should be able to validate if the parameter is from the Color enum. That would avoid the switch and the validateSet train :D but it's fine for now :)

Reply:
You are probably correct.. i was researching the same thing using a color enum, but with the same result. If we could find a way to get RGB565 or RGB888 colors it would probably be dead easy to use it instead. Which enum was it you were looking at? I cant find any way to get RGB values from system.consolecolor (the only one i know of..)